main/lsp/requests/hover.cc

```diff
@@ -89,8 +89,10 @@ unique_ptr<ResponseMessage> HoverTask::runRequest(LSPTypecheckerDelegate &typech
             if (constraint) {
                 retType = core::Types::instantiate(gs, retType, *constraint);
             }
-            if (sendResp->dispatchResult->main.method.exists() && sendResp->dispatchResult->main.method.isSynthetic()) {
-                // For synthetic methods, just show the return type
+            if (sendResp->dispatchResult->main.method.exists() &&
+                sendResp->dispatchResult->main.method.data(gs)->owner == core::Symbols::MagicSingleton()) {
+                // Most <Magic>.<foo> are not meant to be exposed to the user. Instead, just show
+                // the result type.
```

Comment:
What motivated removing the check for `isSynthetic`? Does that not overlap with the magic methods?

Reply:
See this commit:

https://github.com/sorbet/sorbet/pull/3682/commits/dc3d807ae030d5954e6e7a3f10f9fd5165afac06

and the attached commit message:

```
Fix failing LSP test

Now that we included Class#new in the dispatch result, a hover result
changed from `(nothing)` to something, but it looked bad, because
`Class#new` is entered as a "synthetic symbol" (i.e., any method entered
in GlobalState.cc).

That heuristic was too broad for what this code wanted to do, which is
show `{foo: Integer(0)}` when you hover on a hash instead of
`<Magic>.<build-hash>`, etc. (there are a lot `<Magic>.<build-*>`).

So instead, I check whether the owner is `T.class_of(<Magic>)` because
that's a better indication that the thing shouldn't be printed.
```